pkgs/helpers/regex_test.go

```diff
@@ -0,0 +1,47 @@
+package regexhandler_test
+
+import (
+	"testing"
+
+	regexhandler "github.com/Lubwama-Emmanuel/scrapper_golang/pkgs/helpers"
+)
+
+func TestMatchCompanyLink(t *testing.T) {
+	t.Parallel()
+	type test struct {
+		link        string
+		companyLink string
+		companyName string
+	}
+	tests := []test{
+		{"/url?q=https://www.mukwano.com/&sa=U&ved=2ah", "https://www.mukwano.com", "mukwano"},
+		{"/url?q=http://www.mukwano.com/&sa=U&ved=2ahUK", "http://www.mukwano.com", "mukwano"},
+	}
+
+	for i := range tests {
+		got := regexhandler.MatchCompanyLink(tests[i].link, tests[i].companyName)
+
+		if got != tests[i].companyLink {
+			t.Fatalf("Expected: %v But Got: %v", tests[i].companyLink, got)
+		}
+	}
+}
+
+func TestMatchEmail(t *testing.T) {
+	t.Parallel()
+	type test struct {
+		link string
+		want string
+	}
+	tests := []test{
+		{"customercare@mukwano.com", "customercare@mukwano.com"},
+	}
+
+	for i := range tests {
+		got := regexhandler.MatchEmail(tests[i].link)
```

Comment:
You can look into how to use sub test runs to improve this, or remind me to show you how to use  them in our next sync

Reply:
https://github.com/Lubwama-Emmanuel/scrapper_golang/issues/2#issue-1613337287 moved this to an issue and will merge this PR